src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncIteratorTests.cs

```diff
@@ -8672,5 +8673,154 @@ static async System.Threading.Tasks.Task Main()
             comp2.VerifyEmitDiagnostics(); // Indirectly calling IsMetadataVirtual on S.DisposeAsync (a read which causes the lock to be set)
             comp1.VerifyEmitDiagnostics(); // Would call EnsureMetadataVirtual on S.DisposeAsync and would therefore assert if S was not already ForceCompleted
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/68027")]
+        public void LambdaWithBindingErrorInYieldReturn()
+        {
+            var src = """
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+class C
+{
+    static async IAsyncEnumerable<Func<string, Task<string>>> BarAsync()
+    {
+        yield return async s =>
+        {
+            await Task.CompletedTask;
+            throw new NotImplementedException();
+        };
+    }
+}
+""";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80);
+            comp.VerifyDiagnostics(
+                // (7,63): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
```

Comment:
>warning CS1998 [](http://example.com/codeflow?start=27&length=14)

Consider suppressing this warning if it is not a goal of the test to verify its presence